netboot/netconf.go

```diff
@@ -127,5 +216,34 @@ func ConfigureInterface(ifname string, netconf *NetConf) error {
 		resolvconf += fmt.Sprintf("nameserver %s\n", ns)
 	}
 	resolvconf += fmt.Sprintf("search %s\n", strings.Join(netconf.DNSSearchList, " "))
-	return ioutil.WriteFile("/etc/resolv.conf", []byte(resolvconf), 0644)
+	if err = ioutil.WriteFile("/etc/resolv.conf", []byte(resolvconf), 0644); err != nil {
+		return fmt.Errorf("could not write resolv.conf file %v", err)
+	}
+
+	// add default route information for v4 space. only one default route is allowed
+	// so ignore the others if there are multiple ones
+	if len(netconf.Routers) > 0 {
+		iface, err = netlink.LinkByName(ifname)
+		if err != nil {
+			return fmt.Errorf("could not obtain interface when adding default route: %v", err)
+		}
+		// if there is a default v4 route, remove it, as we want to add the one we just got during
+		// the dhcp transaction. if the route is not present, which is the final state we want,
+		// an error is returned so ignore it
+		dst := &net.IPNet{
+			IP:   net.IPv4(0, 0, 0, 0),
+			Mask: net.CIDRMask(0, 32),
+		}
+		route := netlink.Route{LinkIndex: iface.Attrs().Index, Dst: dst, Src: net.IPv4(0, 0, 0, 0)}
```

Comment:
question: does this work also when there are multiple default routes?

Reply:
I haven't honestly tested with multiple default routes. I guess this will delete all default source based and non source based routes. I can test that scenario as well, but in a scenario with multiple routes, this is very intrusive. At the same time, I wouldn't have, off the top of my head, an alternative solution.

Reply:
I don't know the exact solution either, I thought that a loop + check could be enough, but it's probably overcomplicating things. At this point let's leave this as it is, and maybe add a comment about "no idea what happens with multiple default routes". Would that make sense?

Reply:
I'll see at least what happens with multiple default routes and leave a comment behind.

Reply:
I have experimented a bit and the control path above actually removes only the default route (dst 0.0.0.0) to the L2 domain (gw: 0.0.0.0), which is what one would want to add to an unconfigured interface to make DHCP transaction succeed. Also, if this default route has a specific metric assigned, it doesn't get removed. The code doesn't remove any other default route (i.e. gw != 0.0.0.0). The code then adds a default route via the gateway specified in the OFFER for only the SRC address specified in the OFFER. Looks like a sane behavior.

Reply:
I left the above as a comment in the code.